new-lamassu-admin/src/pages/Maintenance/MachineDetailsCard.js

```diff
@@ -69,22 +97,63 @@ const Item = ({ children, ...props }) => (
 )
 
 const MachineDetailsRow = ({ it: machine, onActionSuccess }) => {
-  const [action, setAction] = useState(null)
+  const [action, setAction] = useState({ command: null })
   const [errorMessage, setErrorMessage] = useState(null)
   const classes = useMDStyles()
 
+  const [
+    fetchMachineEvents,
+    { loading: loadingEvents, data: machineEventsLazy }
+  ] = useLazyQuery(MACHINE, {
+    variables: {
+      deviceId: machine.deviceId
+    }
+  })
+
+  useEffect(() => {
+    if (action.command === 'restartServices') {
+      fetchMachineEvents()
+    }
+  }, [action.command, fetchMachineEvents])
+
+  useEffect(() => {
+    if (machineEventsLazy && action.command === 'restartServices') {
+      const state = JSON.parse(
+        machineEventsLazy.machine.latestEvent?.note ?? '{"state": null}'
+      ).state
+      if (!isStaticState(state)) {
+        setAction(action => ({
+          ...action,
+          message: (
+            <span className={classes.warning}>
+              A user may be in the middle of a transaction and they could lose
+              their funds if you continue.
+            </span>
+          )
+        }))
+      } else {
+        // clear message from object when state goes from not static to static
+        setAction(action => ({
+          ...action,
+          message: null
+        }))
+      }
+    }
+  }, [action.command, classes.warning, machineEventsLazy])
```

Comment:
Ok, now it's easier to rework this part of the code. 

- Since machineEventsLazy is a common variable we can skip using useEffect altogether.
- Move the span to a variable
- Since the only difference between the if/else is message, might as well do a ternary if for a message const